.github/workflows/ensure_linting.yml

```diff
@@ -0,0 +1,23 @@
+name: Ensure Linting
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  check_linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b #v27
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: run treefmt in CI mode
+        run: nix flake check --impure
```

Comment:
I imagine `--impure` is a left over from before we defined the `checks` output. Without the `checks` it seems to want top check `devShell`, and that does require `--impure`. So maybe we don't need it here.

Supposedly this PR was to prevent `devShell` requiring `--impure` also,  https://github.com/cachix/devenv/pull/1018 - - I think that requires adding a new input to our flake though (if I understand this correctly).

Reply:
Yeah it was a result of the devShell.

Reply:
Ah bummer. Well if we need it, we need it. I'll just cry every time I see this ;)

Reply:
I believe it shouldn't be required anymore, because the tests don't construct the devenv anymore.